### PR TITLE
Predict random image

### PR DESCRIPTION
**Pull request by tordbb**
Lets the predict() function be used without any inputs. In that case, a randomly selected image of a randomly selected class of the training data is predicted upon.

### CODE REVIEW

File: fruit_classifier/predict/__main__.py [hunk 1]
```diff
@@ -20,6 +21,23 @@ def main(image_path):
         The image path as a string
     """
 
+    if image_path == '':
```
Comment (fabiansd) [inline fruit_classifier/predict/__main__.py:31]: I understand that this function picks a random image from a random folder and predicts on this image with the trained model. 

This means that an image from the training data could be used as input, is this the intention? If you would like to predict on test images you should save the train-test split distribution from the training phase somehow.

**Review by fabiansd — CHANGES_REQUESTED** [review]
This functionality needs a unittest in order to be accepted for merging

File: fruit_classifier/predict/__main__.py (hunk 1 above)
Comment (tordbb) [reply fruit_classifier/predict/__main__.py:31]: Correct, ideally we should only pick images from the validation/test set, not from the training set. Could this be left as an issue/improvement, and merging this increment into master?

File: fruit_classifier/predict/predict_utils.py [hunk 2]
```diff
@@ -1,4 +1,6 @@
 import pickle
+import random
+
```
Comment (loeiten) [inline fruit_classifier/predict/predict_utils.py]: Please avoid adding newlines to the import statements

File: fruit_classifier/predict/predict_utils.py [hunk 3]
```diff
@@ -96,3 +98,33 @@ def load_classifier():
     model = load_model(str(model_path))
 
     return model
+
+
+def pick_random_image(from_folder):
```
Comment (loeiten) [inline fruit_classifier/predict/predict_utils.py]: Recall that "folder" is a windows term, and that "directory" is the preferred term

File: fruit_classifier/predict/predict_utils.py [hunk 4]
```diff
@@ -96,3 +98,33 @@ def load_classifier():
...
+    Note, this includes images from both training set and test set.
+
+    Parameters
+    ----------
+    from_folder : str - the folder where an image is found
+
+    Returns
+    -------
+    image_path : str - the path to the image
+
```
Comment (loeiten) [inline fruit_classifier/predict/predict_utils.py:116]: No newline

File: fruit_classifier/predict/predict_utils.py [hunk 5]
```diff
@@ -96,3 +98,33 @@ def load_classifier():
...
+def pick_random_image(from_folder):
+    """
+    Picks a random image from a random directory in "from_folder"
+
+    Note, this includes images from both training set and test set.
+
+    Parameters
+    ----------
+    from_folder : str - the folder where an image is found
```
Comment (loeiten) [inline fruit_classifier/predict/predict_utils.py]: Please use numpy docstring

```suggestion
    from_folder : str
        The directory where an image is found
```

File: fruit_classifier/predict/predict_utils.py [hunk 6]
```diff
@@ -96,3 +98,33 @@ def load_classifier():
...
+
+    Returns
+    -------
+    image_path : str - the path to the image
```
Comment (loeiten) [inline fruit_classifier/predict/predict_utils.py]: Please use numpy docstring

```suggestion
    image_path : str 
        the path to the image
```

File: fruit_classifier/predict/predict_utils.py (hunk 5 above)
Comment (loeiten) [reply fruit_classifier/predict/predict_utils.py]: Also, it looks like this will be relative to the `generated_data` directory. If so this should be documented

File: fruit_classifier/predict/predict_utils.py [hunk 7]
```diff
@@ -96,3 +98,33 @@ def load_classifier():
...
+
+    """
+    generated_data_dir = \
+        Path(__file__).absolute().parents[2].joinpath(
+            'generated_data')
+    cleaned_dir = generated_data_dir.joinpath(from_folder)
```
Comment (loeiten) [inline fruit_classifier/predict/predict_utils.py]: Is there a reason why this is called `cleaned_dir`? It could be misleading when reading the code when the input is unknown

**Review by loeiten — CHANGES_REQUESTED** [review]
Apart from the minor comments, this is valid code.

However, I'm uncertain of the user need for this. Around 50 % of functionality in software is rarely or never used, and I suspect that this is one of these functionality. The drawback by having unused code (apart from being unused) is that is carries with it future need of maintenance.

However, I can be heavily blinded by my own beliefs, so if @fabiansd also think that this is valuable, it's fine with me to include it

File: fruit_classifier/predict/predict_utils.py (hunk 3 above)
Comment (tordbb) [reply fruit_classifier/predict/predict_utils.py]: Replaced folder with directory.

File: fruit_classifier/predict/predict_utils.py (hunk 7 above)
Comment (tordbb) [reply fruit_classifier/predict/predict_utils.py]: Changed to "image_dir".

Comment (tordbb) [reply fruit_classifier/predict/predict_utils.py]: Also, slightly simplified code of that function.
